lib/fog/openstack/identity/v3/requests/create_application_credentials.rb

```diff
@@ -0,0 +1,23 @@
+module Fog
+    module OpenStack
+      class Identity
+        class V3
+          class Real
+            def create_application_credentials(credential = {})
+              user_id = credential.delete('user_id') || credential.delete(:user_id)
+              puts Fog::JSON.encode(:application_credential => credential)
+              request(
+                :expects => [201],
+                :method  => 'POST',
+                :path    => "users/#{user_id}/application_credentials",
+                :body    => Fog::JSON.encode(:application_credential => credential)
+              )
+            end
+          end
+  
+          class Mock
```

Comment:
For consistency with other parts of fog, could you add a method to each of the mock versions that has the same signature, and whose body is:
```
raise Fog::Errors::MockNotImplemented
```

That way it helps ensure that the behavior is obvious/consistent, as opposed to potentially NoMethod errors or even silently failing if the method is defined without a body as below.